xbmc/platform/win32/WinMain.cpp

```diff
@@ -84,26 +99,43 @@ INT WINAPI WinMain(HINSTANCE hInst, HINSTANCE, LPSTR commandLine, INT)
   using KODI::PLATFORM::WINDOWS::ToW;
   std::string appName = CCompileInfo::GetAppName();
   HANDLE appRunningMutex = CreateMutex(nullptr, FALSE, ToW(appName + " Media Center").c_str());
-  if (GetLastError() == ERROR_ALREADY_EXISTS)
+  if (appRunningMutex == nullptr)
   {
-    auto appNameW = ToW(appName);
-    HWND hwnd = FindWindow(appNameW.c_str(), appNameW.c_str());
-    if (hwnd != nullptr)
+    if (GetLastError() == ERROR_ALREADY_EXISTS)
     {
-      // switch to the running instance
-      ShowWindow(hwnd, SW_RESTORE);
-      SetForegroundWindow(hwnd);
+      auto appNameW = ToW(appName);
+      HWND hwnd = FindWindow(appNameW.c_str(), appNameW.c_str());
+      if (hwnd != nullptr)
+      {
+        // switch to the running instance
+        ShowWindow(hwnd, SW_RESTORE);
+        SetForegroundWindow(hwnd);
+      }
+      return 0;
+    }
+    else
+    {
+      LogError(L"unable to create unique instance mutex");
+      return -1;
     }
-    ReleaseMutex(appRunningMutex);
```

Comment:
When exit early (because another instance is already running) is correct ReleaseMutex